validator/main.go

```diff
@@ -641,19 +734,20 @@ func (d *Driver) validate() error {
 		return err
 	}
 
-	// delete host driver status file is already present
-	err = deleteStatusFile(outputDirFlag + "/" + hostDriverStatusFile)
-	if err != nil {
-		return err
-	}
-
 	driverRoot, isHostDriver, err := d.runValidation(false)
 	if err != nil {
 		log.Error("driver is not ready")
 		return err
 	}
 
-	if !disableDevCharSymlinkCreation {
+	if disableDevCharSymlinkCreation { //nolint:gocritic
```

Comment:
Let's rather factor this into a function.

Reply:
Done